web/profiles/custom/yalesites_profile/modules/custom/ys_templated_content/src/Modifiers/TemplateModifier.php

```diff
@@ -0,0 +1,228 @@
+    return new static(
+      $container->get('uuid'),
+      $container->get('path_alias.repository'),
+    );
+  }
+
+  /**
+   * Process the content array.
+   *
+   * @param array $content_array
+   *   The content array.
+   */
+  public function process($content_array) {
+    $originalUuid = $content_array['uuid'];
+    $newUuid = $this->uuidService->generate();
+    $content_array['base_fields']['created'] = $this->getUnixTimestamp();
+    $content_array = $this->replaceBrokenImages($content_array);
+    $content_array = $this->generateAlias($content_array);
+    $content_array = $this->replaceUuids($content_array, $originalUuid, $newUuid);
+
+    return $content_array;
+  }
+
+  /**
+   * Replace broken images with a placeholder.
+   *
+   * @param array $content_array
+   *   The content array.
+   *
+   * @return array
+   *   The content array with images fixed with placeholder.
+   */
+  protected function replaceBrokenImages(array $content_array) : array {
+    foreach ($content_array as $key => $value) {
+      if (is_array($value)) {
+        $content_array[$key] = $this->replaceBrokenImages($value);
+      }
+      elseif ($key == 'uri' && strpos($value, 'public://') !== FALSE) {
+        $path = $value;
+        $path = str_replace('public://', 'sites/default/files/', $path);
+        if (!file_exists($path)) {
+          $content_array[$key] = $this::PLACEHOLDER;
+        }
+      }
+    }
+
+    return $content_array;
+  }
+
+  /**
+   * Replace the UUIDs in the content array.
+   *
+   * Any reference to the main UUID should be replaced so that we reference
+   * the newly created one.
+   *
+   * @param array $content
+   *   The content array.
+   * @param string $uuid
+   *   The UUID to replace.
+   * @param string $newUuid
+   *   The new UUID.
+   *
+   * @return array
+   *   The content array with the UUIDs replaced.
+   */
+  public function replaceUuids($content, $uuid, $newUuid) {
+    if (array_key_exists('uuid', $content) && $content['uuid'] === $uuid) {
+      $content['uuid'] = $newUuid;
+    }
+    // Find any other element that has the original UUID passed and replace it
+    // with the new one.
+    foreach ($content as $key => $value) {
+      if (is_array($value)) {
+        $content[$key] = $this->replaceUuids($value, $uuid, $newUuid);
+      }
+      elseif (is_string($value) && strpos($value, 'entity:node/') !== FALSE) {
+        $content[$key] = str_replace('entity:node/' . $uuid, 'entity:node/' . $newUuid, $value);
+      }
+      elseif (is_string($value) && strpos($value, $uuid) !== FALSE) {
+        $content[$key] = str_replace($uuid, $newUuid, $value);
+      }
+    }
+
+    return $content;
+  }
+
+  /**
+   * Generate a unique alias using the current date/time.
+   *
+   * @param string $alias
+   *   The alias.
+   *
+   * @return string
+   *   The unique alias (original alias with date/time).
+   */
+  protected function generateUniqueAliasWithDate($alias) {
```

Comment:
This method is not currently in use.

Reply:
That's for reminding me about that one--that was an alternative to the generation of an alias that I knew wouldn't be taken.  Now it goes sequentially as this one seemed like it could puzzle users more than the 1, 2, 3 aspect.